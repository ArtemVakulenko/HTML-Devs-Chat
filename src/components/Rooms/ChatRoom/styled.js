import styled from 'styled-components';

export const StChatRoomContainer = styled.div`
  padding: 10px;
  display: grid;
  grid-template-columns: 20% 70% 10%;
  align-items: flex-end;
  cursor: pointer;
  background: ${({ colors, theme, isActive }) => (isActive ? colors[theme].button : 'inherit')};
  &:hover{
    background: ${({ colors, theme, isActive }) => (isActive ? colors[theme].button : colors[theme].input)};
  }
`;

export const StChatImg = styled.img`
  width: 50px;
  max-height: 50px;
`;

export const StInfo = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 12px;
`;

export const StMessageCount = styled.div`
  display: flex;
  border-radius: 100%;
  background: #5C9DDD;
  align-items: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  color: black;
  font-size: 12px;
`;
