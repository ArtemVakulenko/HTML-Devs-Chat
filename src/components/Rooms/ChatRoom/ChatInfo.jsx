import React from 'react';
import PropTypes from 'prop-types';
import { StInfo, StMessageCount } from './styled.js';

const ChatInfo = ({ unreadCount }) => {
  return (
    <StInfo >
      {
        unreadCount
          ? <StMessageCount><span>{unreadCount}</span></StMessageCount>
          : null
      }
    </StInfo>
  );
};

ChatInfo.propTypes = {
  unreadCount: PropTypes.number,
};

export default ChatInfo;
