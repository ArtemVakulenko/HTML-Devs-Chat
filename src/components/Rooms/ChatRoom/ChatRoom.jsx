import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { useTheme } from '../../hooks';
import Title from '../../_common/Title/Title';
import ChatInfo from './ChatInfo';
import { StChatRoomContainer, StChatImg } from './styled';

const ChatRoom = ({
  id,
  img,
  title,
  match,
  unreadCount,
  setActiveRoom,
  clearMessageValue,
  clearUnreadMessages,
}) => {
  const { colors, theme } = useTheme();
  const handleSetActiveRoom = () => {
    setActiveRoom(id);
    clearUnreadMessages(id);
    clearMessageValue();
  };
  useEffect(() => {
    if (match.params.id) {
      const roomId = match.params.id;
      setActiveRoom(roomId);
    }
  }, []);
  return (
    <NavLink to={`/main/${id}`} onClick={handleSetActiveRoom}>
      <StChatRoomContainer
        theme={theme}
        colors={colors}
        isActive={match.params.id === id}
      >
        <StChatImg src={img || '../assets/img/iconfinder-icon.svg'} />
        <Title title={title} isTranslate={false} />
        <ChatInfo unreadCount={unreadCount} />
      </StChatRoomContainer>
    </NavLink>
  );
};

export default ChatRoom;

ChatRoom.propTypes = {
  id: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  match: PropTypes.object,
  unreadCount: PropTypes.number,
  setActiveRoom: PropTypes.func,
  clearMessageValue: PropTypes.func,
  clearUnreadMessages: PropTypes.func,
};
