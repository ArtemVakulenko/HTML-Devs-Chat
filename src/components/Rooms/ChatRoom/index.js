import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { clearUnreadMessages, clearMessageValue } from 'store/mainPage/actions';
import ChatRoom from './ChatRoom';

const mapDispatchToProps = {
  clearUnreadMessages, clearMessageValue,
};
export default compose(
  connect(null, mapDispatchToProps),
  withRouter,
)(ChatRoom);
