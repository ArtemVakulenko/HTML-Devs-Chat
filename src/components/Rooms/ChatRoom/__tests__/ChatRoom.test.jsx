import { mount, shallow } from 'enzyme';
import React from 'react';
import ChatRoom from '../ChatRoom';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
        button: '',
      },
    },
    theme: 'dark',
  })),
}));
jest.mock('react-router-dom', () => ({
  NavLink: jest.fn()
    .mockImplementation(
      (props) => <a href={props.path} >{props.children}</a>,
    ),
}));

describe('ChatRoom', () => {
  const props = {
    id: '123',
    img: 'src',
    title: 'title',
    match: { params: { id: '123' } },
    unreadCount: 1,
    setActiveRoom: jest.fn(),
    clearMessageValue: jest.fn(),
    clearUnreadMessages: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('Should match snapshot', () => {
    const component = shallow(<ChatRoom {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have a', () => {
    const component = mount(<ChatRoom {...props} />);
    expect(component.find('a')).toHaveLength(1);
  });
  it('Should have Title', () => {
    const component = mount(<ChatRoom {...props} />);
    expect(component.find('Title')).toHaveLength(1);
  });
  it('Should have StChatRoomContainer', () => {
    const component = mount(<ChatRoom {...props} />);
    expect(component.find('styled__StChatRoomContainer')).toHaveLength(1);
  });
  it('Should have StChatImg', () => {
    const component = mount(<ChatRoom {...props} />);
    expect(component.find('styled__StChatImg')).toHaveLength(1);
  });
  it('Should have ChatInfo', () => {
    const component = mount(<ChatRoom {...props} />);
    expect(component.find('ChatInfo')).toHaveLength(1);
  });
  it('Should have call handlesetActive room && setActiveRoom', () => {
    const component = mount(<ChatRoom {...props} />);
    component.find('a').simulate('click');
    expect(component.props().setActiveRoom).toHaveBeenCalledWith(props.id);
  });
  it('Should have call handlesetActive room && clearUnreadMessages', () => {
    const component = mount(<ChatRoom {...props} />);
    component.find('a').simulate('click');
    expect(component.props().setActiveRoom).toHaveBeenCalledWith(props.id);
  });
  it('Should have call handlesetActive room && clearMessageValue', () => {
    const component = mount(<ChatRoom {...props} />);
    component.find('a').simulate('click');
    expect(component.props().setActiveRoom).toHaveBeenCalled();
  });
  it('Should have matchparams id', () => {
    const component = mount(<ChatRoom {...props} />);
    expect(component.props().setActiveRoom).toHaveBeenCalledTimes(1);
  });
  it('Should have !matchparams id', () => {
    props.match.params.id = '';
    const component = mount(<ChatRoom {...props} />);
    expect(component.props().setActiveRoom).not.toHaveBeenCalled();
  });
});
