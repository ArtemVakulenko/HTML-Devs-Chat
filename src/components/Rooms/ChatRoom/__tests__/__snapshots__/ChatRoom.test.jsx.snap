// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatRoom Should match snapshot 1`] = `"<a><div class=\\"styled__StChatRoomContainer-fFLKRu dojhib\\"><img src=\\"src\\" class=\\"styled__StChatImg-kbqTCl hBkeLG\\"/><h1 class=\\"styled__AppTitle-bMNsHC jWDbjD\\">title</h1><div class=\\"styled__StInfo-ckAQui cfpCEt\\"><div class=\\"styled__StMessageCount-fPOoeU cQHHxW\\"><span>1</span></div></div></div></a>"`;
